internal/ir/xds.go

```diff
@@ -223,8 +223,20 @@ type CoreListenerDetails struct {
 	ExtensionRefs []*UnstructuredRef `json:"extensionRefs,omitempty" yaml:"extensionRefs,omitempty"`
 	// Metadata is used to enrich envoy resource metadata with user and provider-specific information
 	Metadata *ResourceMetadata `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	// IPFamily specifies the IP address family for the gateway.
+	// It can be IPv4, IPv6, or Dual.
+	IPFamily IPFamily `json:"ipFamily,omitempty" yaml:"ipFamily,omitempty"`
```

Comment:
my preference would be to not use IPFamily in this layer but convert the `Address` field into a list and pass `::` it and in the xds layer iterate deal with this list and convert it into `Address` and `AdditionalAddresses` fields